spacy/vocab.pyx

```diff
@@ -332,8 +332,8 @@ cdef class Vocab:
         remap = {}
         for i, key in enumerate(keys[nr_row:]):
             self.vectors.add(key, row=syn_rows[i][0])
-            word = self.strings[key]
-            synonym = self.strings[syn_keys[i][0]]
+            word = self.strings[int(key)]
```

Comment:
If this is designed to copy data from cupy to numpy, then this will be very slow, because we'll be syncing each element one-by-one. We should call `keys.get()` on the whole array (or slice that we need) before the loop.

Reply:
I need to dive into the detail again, but I think the `int()` bit was required when running one of the tests from Adriane's PR On GPU, because `key` would be a cupy array with just one element that doesn't get automatically converted to a singe element like numpy does.